tripal_chado_views/tripal_chado_views.views.inc

```diff
@@ -600,84 +600,97 @@ function tripal_chado_views_views_data_tripal_views_tables($data) {
-            'field' => $base_table . '_id',
-          );
-          $data['node']['table']['join'][$base_table] = array(
-            'left_table' => $linker_table,
-            'left_field' => 'nid',
-            'field' => 'nid',
-          );
-
-          // Adds in a chado base table => node relationship
-          // This allows controlled joining to multiple nodes per line
-          // Use Case:  link to feature and organism nodes on a feature listing
-          // D7 todo: a custom relationship handler to get from feature.organism_id => organism node
-          //      without 1st needing to add relationship to organism table
-          $base_field = $base_table . '_id';
-          $data[$linker_table][$base_field] = array(
-            'group' => $base_title,
-            'title' => $base_title . 'Node',
-            'help' => t("Links @base_title to it's node.", array('@base_title' => $base_title)),
-            'relationship' => array(
-              'handler' => 'views_handler_relationship',
-              'title' => t("@base_title => Node", array('@base_title' => $base_title)),
-              'label' => t("@base_title => Node", array('@base_title' => $base_title)),
-              'real field' => 'nid',
-              'base' => 'node',
-              'base field' => 'nid'
-            ),
-          );
-
-          // Add Chado fields to a node-based view
-          // This will only be done with relationships
-          $base_field = $base_table . '_id';
-          $data['node'][$base_field] = array(
-            'group' => $base_title,
-            'title' => $base_title,
-            'help' => t("Links node to chado @base_title.", array('@base_title' => $base_title)),
-            'relationship' => array(
-              'handler' => 'views_handler_relationship',
-              'title' => t("Node => @base_title", array('@base_title' => $base_title)),
-              'label' => t("Node => @base_title", array('@base_title' => $base_title)),
-              'real field' => 'nid',
-              'base' => $linker_table,
-              'base field' => 'nid'
-            ),
-          );
-
-        }
+  // ADD IN ENTITIES JOINS & RELATIONSHIPS
+  // D7 @todo: Create custom handler to allow join from Entity => Base (ie: organism)
+  //           with the addition of a single relationship
+  // D7 @todo: Create custom handler to allow join from Base (ie: organism)
+  //           with the addition of a single relationship
+  // D7 @todo: Add support for Mview <-> Entity joins and relationships
+  $bundles = db_select('chado_bundle', 'CB')
+      ->fields('CB', array('bundle_id', 'data_table', 'type_column', 'type_id'))
+      ->execute();
+  // Look for the record ID in the appropriate chado table.
+  foreach ($bundles as $bundle) {
+    // Ids we'll use for queries.
+    $base_table = $bundle->data_table;
+    $linker_table = 'chado_bio_data_' . $bundle->bundle_id;
+    $base_title = ucwords(str_replace('_', ' ', $base_table));
```

Comment:
This needs to be the bundle name. Currently I end up with 4 different `Stock: Stock => Entity` relationships and I need to use trial and error to determine which one is right for my content. This is also only a good solution if all my content is of one Tripal Content Type. Otherwise, it becomes difficult since you need to use a conditional to set a link on the stock name.

This is what I currently see in my relationship list on a fresh Tripal install:
![screen shot 2019-03-06 at 7 12 13 pm](https://user-images.githubusercontent.com/1566301/53925206-b9c28e80-4044-11e9-9e6e-26fcf8408f12.png)

I was able to confirm that one I selected the right `Stock: Stock => Entity` for my bundle that I got the entity links 🎉

Reply:
It's a bit weird because I had only "Stock: Stock => Entity" on my system but I re-installed everything and restarted from scratch and that's right, 4 relationships appeared! :-/ That's not what's wanted. I'll try see what's going on.